extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java

```diff
@@ -0,0 +1,93 @@
+package io.quarkus.vertx;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+import io.quarkus.vertx.core.runtime.VertxCoreRecorder;
+import io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle;
+import io.smallrye.common.vertx.VertxContext;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+/**
+ * Provides utility methods to work with Vertx duplicated context.
+ *
+ * @see VertxContext
+ */
+public final class VertxContextSupport {
+
+    private VertxContextSupport() {
+    }
+
+    /**
+     * Subscribes to the supplied {@link Uni} on a Vertx duplicated context; blocks the current thread and waits for the result.
+     *
+     * @param uniSupplier
+     * @throws Throwable
+     */
+    public static <T> T subscribeAndBlock(Supplier<Uni<T>> uniSupplier) throws Throwable {
+        T ret = null;
+
+        Context context = Vertx.currentContext();
+        if (context == null) {
+            Vertx vertx = VertxCoreRecorder.getVertx().get();
+            context = VertxContext.getOrCreateDuplicatedContext(vertx);
+        } else {
+            // Executed on a vertx thread
```

Comment:
Maybe we should verify here that the Vert.x thread we're running on is not an event loop thread?

Reply:
So that we don't block the event loop, right? In general, this method should not be called on an event loop so yes, we could even throw an exception...

Reply:
Exception is exactly what I had in mind, yes.

Reply:
Done.